### PR TITLE
conditionals-simple_programs_with_decision_points-medium-q4

### PR DESCRIPTION
**Pull request by Hedgemon4**

### CODE REVIEW

**Review by SamamthaH — COMMENTED** [review]
From what I know, they're going to fix the integers not taking negatives, so you might want to change your code when that patch comes out. Aside from that it looks good.

**Review by SanaBob — DISMISSED** [review]

File: src/conditionals/simple_programs_with_decision_points/medium/q4/MainTest.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    static Stream<Arguments> inputProvider(){
+        return Stream.of(Arguments.of(-20, -40, 20, -40), Arguments.of(49, 10, 39, 10), Arguments.of(20, 50, 20, 50), Arguments.of(500, 0, 500, 0), Arguments.of(-40, -20, -40, -20));
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    void inputTest(int inputA, int inputB, int outputA, int outputB){
+        runWithInput(inputA + System.lineSeparator() + inputB);
+        assertEquals(Integer.parseInt(getItemByName("a")), outputA, "Calculated value of A is incorrect. Make sure a = a - b if a > b.");
+        assertEquals(Integer.parseInt(getItemByName("b")), outputB, "Calculated value of B is incorrect.");
```
Comment (Opeyem1a) [inline src/conditionals/simple_programs_with_decision_points/medium/q4/MainTest.java:46]: "The calculated..." for both

File: src/conditionals/simple_programs_with_decision_points/medium/q4/MainTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    static Stream<Arguments> inputProvider(){
+        return Stream.of(Arguments.of(-20, -40, 20, -40), Arguments.of(49, 10, 39, 10), Arguments.of(20, 50, 20, 50), Arguments.of(500, 0, 500, 0), Arguments.of(-40, -20, -40, -20));
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    void inputTest(int inputA, int inputB, int outputA, int outputB){
+        runWithInput(inputA + System.lineSeparator() + inputB);
+        assertEquals(Integer.parseInt(getItemByName("a")), outputA, "Calculated value of A is incorrect. Make sure a = a - b if a > b.");
```
Comment (Opeyem1a) [inline src/conditionals/simple_programs_with_decision_points/medium/q4/MainTest.java:45]: Should we be helping them this much though? @aadiraju what do you think?

**Review by Opeyem1a — CHANGES_REQUESTED** [review]

**Review by SamamthaH — DISMISSED** [review]

File: src/conditionals/simple_programs_with_decision_points/medium/q4/MainTest.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                new IntegerLiteral("b")
+        };
+    }
+
+    public void runMain() {
+        TheBiggerInt.main(new String[0]);
+    }
+
+    static Stream<Arguments> inputProvider() {
+        return Stream.of(Arguments.of(-20, -40, 20, -40), Arguments.of(49, 10, 39, 10), Arguments.of(20, 50, 20, 50),
```
Comment (Opeyem1a) [inline src/conditionals/simple_programs_with_decision_points/medium/q4/MainTest.java:41]: What about the case where both numbers are equal?

File: src/conditionals/simple_programs_with_decision_points/medium/q4/MainTest.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+                Arguments.of(500, 0, 500, 0), Arguments.of(-40, -20, -40, -20));
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    void correctNumberOutput(int inputA, int inputB, int outputA, int outputB) {
+        runWithInput(inputA + System.lineSeparator() + inputB);
+        assertEquals(Integer.parseInt(getItemByName("a")), outputA, "The calculated value of A is incorrect.");
```
Comment (Opeyem1a) [inline src/conditionals/simple_programs_with_decision_points/medium/q4/MainTest.java:49]: Would this error message be meaningful to a student completing the question? If you think so then all good

Comment (Opeyem1a) [reply src/conditionals/simple_programs_with_decision_points/medium/q4/MainTest.java:49]: Even if the student knows variable `a`, then it shouldn't be uppercase here

**Review by Opeyem1a — CHANGES_REQUESTED** [review]

File: src/conditionals/simple_programs_with_decision_points/medium/q4/MainTest.java (hunk 4 above)
Comment (Hedgemon4) [reply src/conditionals/simple_programs_with_decision_points/medium/q4/MainTest.java:49]: I added a note to the question specifying that the first int is a and the second int is b

**Review by SamamthaH — APPROVED** [review]

**Review by Opeyem1a — APPROVED** [review]

**Pull request merged**